File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
        if(needle.length() > haystack.length()) return -1;
        for(int i=0; i<haystack.length(); i++) {
            if(haystack.charAt(i) == needle.charAt(0) 
                && i+needle.length() <= haystack.length()
                && haystack.substring(i, i+needle.length()).equals(needle)) {
                    return i;
                }
        }
        return -1;
    }
}

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.ts
function strStr(haystack: string, needle: string): number {
    const nlen = needle.length
    for(let i=0; i<haystack.length; i++) {
        if(haystack.charAt(i) === needle.charAt(0) && haystack.slice(i, i+nlen) === needle) {
            // console.log("mmm: ", haystack.slice(i, i+nlen))
            return i
        }
    }
    return -1
};